release.md

```diff
@@ -0,0 +1,298 @@
+**DO NOT MERGE** - this is here so people can do review stuff to it if they want to.
+
+⚠️ **Early draft. Details subject to change.** ⚠️
+
+# 🐘 🐘 V5 🐘 🐘  🐘
+
+## 1. All SOLC.
+
+It's now possible to have truffle compile with:
++ any solc-js version listed at [solc-bin](http://solc-bin.ethereum.org/bin/list.json). Specify the one you want and truffle will get it for you.
++ a natively compiled solc binary (you'll need to install this yourself, links to help below).
++ dockerized solc from one of images published [here](https://hub.docker.com/r/ethereum/solc/tags/). (You'll also need to pull down the docker image yourself but it's really easy.)
+
+### List available versions at the command line
+```shell
+$ truffle compile --list                   # Recent stable releases from solc-bin (JS)
+$ truffle compile --list prereleases       # Recent prereleases from solc-bin (JS)
+$ truffle compile --list docker            # Recent docker tags from hub.docker.com
+$ truffle compile --list releases --all    # Complete list of stable releases.
+```
+
+### Specify a solcjs version in `truffle.js`
+
+Set the compilers solc `version` key to the one you'd like. Truffle will fetch it from the solc-bin server and cache it in your local evironment.
+```javascript
+module.exports = {
+  networks: {
+    ... etc ...
+  },
+  compilers: {
+     solc: {
+       version: <string>  // ex:  "0.4.20". (Default: truffle's installed solc)
+     }
+  }
+};
+```
+
+### Advanced:
++ docker
```

Comment:
note for full release: this a great list of things we'll have to remember to document. Making an issue for full-relesae checklist #1176 so we can add stuff as we think of it.